### ISSUE TITLE
bin/setup-composer-auth does not persist auth creds

### ISSUE THREAD

**Issue by markshust** [markshust/docker-magento#567]
**Description**
Running `bin/setup-composer-auth` does not persist Magento's username/password to `var/composer_home/auth.json`, and also causes a file descriptor error on Apple Silicon.

**Steps To Reproduce**
1. Run `bin/setup-composer-auth` and enter in any Composer creds

**Expected Result**
`bin/cli cat var/composer_home/auth.json` outputs stored Composer values

**Actual Result**
`bin/cli cat var/composer_home/auth.json` outputs:

```
{}
```

On Apple Silicone, `bin/setup-composer-auth` also outputs an error: `fork/exec /usr/local/bin/com.docker.cli: bad file descriptor`.

I believe the line `echo "composer config --global http-basic.repo.magento.com ${PUBLIC_KEY} ${PRIVATE_KEY}" | bin/clinotty bash -` is causing the issue.

Perhaps we should detect if a file is located at `~/.composer/auth.json` and link it to `./auth.json`? How does `auth.json` in the project root correlate to `var/composer_home/auth.json` -- what's the fallback/precedence?

**Comment by markshust** [comment]
I may have found something out related to this. Running `bin/setup-composer-auth` returns an error that says:

```
cp: /var/www/.composer/auth.json: No such file or directory
```

...even though that file exists, and is very clearly mounted as a host volume in the `docker-compose.yml` file.

![Screen Shot 2021-11-06 at 11 05 36 AM](https://user-images.githubusercontent.com/437029/140614337-00802413-f6a7-476c-9bd5-db73ad81a346.png)

**Comment by markshust** [comment]
Found a formatting issue, fixed in commit referenced above. Will try to do some additional testing to confirm this resolves the issue.

**Comment by markshust** [comment]
Fixed in `41.0.1`
